sharedbuffers/mapped_struct.py

```diff
@@ -1076,19 +1077,43 @@ def __len__(self):
         else:
             return self.bitlen
 
+    @cython.ccall
+    @cython.locals(dcode=cython.char, offset=cython.size_t)
+    def _contains(self, elem, dcode, offset):
+        xlen = len(self.objlist)
+        hint = xlen // 2
+        pindex = cython.cast(cython.p_char, self.objlist.pybuf.buf) + offset
+        if dcode == 'q':
+            return _c_search_hkey_i64(elem, pindex, 8, xlen, hint)
+        elif dcode == 'I':
+            return _c_search_hkey_ui32(elem, pindex, 4, xlen, hint)
+        elif dcode == 'i':
+            return _c_search_hkey_i32(elem, pindex, 4, xlen, hint)
+        elif dcode == 'd':
+            return _c_search_hkey_f64(elem, pindex, 8, xlen, hint)
+        else:
+            raise NotImplementedError("invalid frozenset data type %s" % chr(dcode))
+
     @cython.locals(i1=cython.Py_ssize_t, i2=cython.Py_ssize_t,
-        step=cython.Py_ssize_t, c=cython.int)
+        step=cython.Py_ssize_t, c=cython.int, dcode=cython.char,
+        offset=cython.size_t)
     def __contains__(self, elem):
         if self.objlist is None:
             try:
                 eint = int(elem)
                 if eint != elem or eint >= 128 or eint < 0:
                     return False
                 return (eint < 64 and (self.bitrep_lo & (1 << eint)) != 0) or (
-                    (eint < 128 and (self.bitrep_hi & (1 << eint)) != 0))
+                    (eint < 128 and (self.bitrep_hi & (1 << (eint - 64))) != 0))
             except (ValueError, TypeError):
                 return False
 
+        if cython.compiled:
+            dcode, _, _, offset, _ = self.objlist._metadata()
+            if dcode not in ('H', 'h', 't'):
+                i1 = self._contains(elem, dcode, offset)
```

Comment:
Why doesn't _contains return a boolean?

Reply:
I was planning on using it in ```subset```, to implement skip search, as you suggested.